packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+  - [FAST](https://explore.fast.design/components/fast-dialog)
+  - [Material UI](https://mui.com/components/dialogs/)
+  - [Radix](https://www.radix-ui.com/docs/primitives/components/dialog)
+  - [Reach UI](https://reach.tech/dialog/)
+  - [Reakit](https://reakit.io/docs/dialog/)
+    - [Ariakit](https://github.com/reakit/reakit/tree/v2)
+  - [Spectrum](https://react-spectrum.adobe.com/react-spectrum/Dialog.html)
 
-## Sample Code
+### Comparison between v0 and v8
+
+Note that the below code samples are not meant to be complete, but to highlight differences between the two libraries. Please refer to official docsites for actual API references.
+
+#### v8
+
+In v8 there are Dialog and Modal components which are relevant to the Dialog component for v9. The Dialog component was intended to be used primarily for confirming actions, whereas Modal component was intended to be used for lengthy content that may contain forms and other controls. This spec will only cover the comparison to the Dialog component.
+
+The visibility of the dialog is controlled through the `hidden` prop whose its value should be a react state boolean provided from the consumer.
+
+[Documentation for v8 Dialog](https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog)
+
+Sample code:
+
+```jsx
+  <DefaultButton secondaryText="Opens the Sample Dialog" onClick={toggleHideDialog} text="Open Dialog" />
+
+  <Dialog
+    hidden={hideDialog}
+    onDismiss={toggleHideDialog}
+    dialogContentProps={dialogContentProps}
+    modalProps={modalProps}
+  >
+    <DialogFooter>
+      <PrimaryButton onClick={toggleHideDialog} text="Send" />
+      <DefaultButton onClick={toggleHideDialog} text="Don't send" />
+    </DialogFooter>
+  </Dialog>
+
+```
+
+#### v0
+
+In v0, the Dialog component expects all the content through props, including the content, actions etc. The dialog component uses the `trigger` prop that expects a React component to control its visibility. The element passed to this prop will be rendered in-place where the dialog is defined.
+
+[Documentation for v0 Dialog](https://fluentsite.z22.web.core.windows.net/components/dialog/definition)
+
+```jsx
+<Dialog
+  cancelButton="Connect protocol"
+  confirmButton="Transmit capacitor"
+  content="Connect driver"
+  header="Transmit capacitor"
+  headerAction="Generate protocol"
+  trigger={<Button content="A trigger" />}
+/>
+```
+
+## API proposal
+
+Dialog component is composed by 3 sub-components: Header, Body and Footer.
+
+### Components
+
+| Component    | Purpose                                                  |
+| ------------ | -------------------------------------------------------- |
+| Dialog       | The main wrapper component.                              |
+| DialogHeader | (optional) Component for the title and the close button. |
+| DialogBody   | (optional) Component for the main content of dialog.     |
+| DialogFooter | (optional) Component for the main actions of dialog.     |
```

Comment:
If we move the close button into its own slot, does the header/body/footer offer anything other than a container, where the body has flex grow? i.e. are there any special aria roles they provide, or custom styling to the children content?

With our composition model, should dialog simply be an empty container, and our examples show proper use for dialogs with title, body, button placement?

Reply:
Yeah, i just want to challenge the inclusion of sections to ensure that there is enough value added by them to justify the cost. If we can reproduce all of the 'value' of these 3 controls in our demos with a tiny bit of boilerplate, we could show best practices, keep the component flexible, and reduce code complexity and improve perf.

Just keep that in mind going forward.